### ISSUE TITLE
Allow ACME to request non-ejabberd hosted subdomains

### ISSUE THREAD

**Issue by licaon-kter** [processone/ejabberd#3075]
The current implementation just uses ACME to get certs for: `domain.tld`, `upload.domain.tld`, `conference.domain.tld`, `proxy.domain.tld`, `pubsubdomain.tld` which, I guess, are extracted from the currently hosted ones by ejabberd.

ejabberd, at least in my case, is not just a single daemon for everything on this domain, I might have others too in need of certs, like `web.domain.tld` (hosting Converse.js on nginx) or anything else really.

So I am forced to add some other [ACME tool](https://letsencrypt.org/docs/client-options/) to generate some certs that might not share the same domain key, so basically different certs with different SANs and expiration dates, on the same base domain.

Now, I do see the point of keeping it strictly related to ejabberd, but the whole thing is rather odd:
* If I use [dehydrated](https://github.com/lukas2511/dehydrated) for `web.domain.tld` why should I bother with the included ACME? I can use this for the ejabberd domains too, right?
* If ejabberd is getting certs for its hosted domains only, why does it generate a cert for `upload.domain.tld` even if that is NOT hosted by itself (`put_url` points to a nginx instance, and with `external_secret` via [ngx_http_upload](https://github.com/weiss/ngx_http_upload)) ?

Maybe add an option `addtional_domains:` where the admin can, besides whatever ejabberd is setup, add other domains.

Right now this feels artificially limited for some reason.

**Comment by zinid** [comment]
I don't think we need to overcomplicate ACME support, because the main goal of built-in ACME client is to simplify ejabberd configuration for newcomers and for admins with simple setups (e.g. XMPP server only - I have such a setup for my personal server). Thus ejabberd only cares about the domains it serves by itself. Your example with the upload component is an exception - ejabberd's ACME client doesn't know anything about modules and `external_secret` option of `mod_http_upload` in particular.

Anyway, I keep the issue opened for a while to hear other people's opinions.

**Comment by licaon-kter** [comment]
> If ejabberd is getting certs for its hosted domains only, why does it generate a cert for upload.domain.tld even if that is NOT hosted by itself (put_url points to a nginx instance, and with external_secret via ngx_http_upload) ?

A simple `if external_secret then exclude upload domain` can be used, imho.

**Comment by Neustradamus** [comment]
It is a really missing point.
To manage all in one place...

**Comment by psjbeisler** [comment]
I was looking at this as well specifically for the converse module, but also BOSH.
Basically just an http domain, cert. the http_upload component could be renamed from `upload` to preferred subdomain (`xmpp`) just to get added to the cert but then it just looks funny in service discovery.
If ejabberd can extend http services (host-meta/bosh) why should it not be able to extend cert issuance to those http as well?
If it could just pick up/add the domains from `hosts_alias:` as well, that feels perfect really.

Honestly, that was my expectation a little bit using the new `hosts_alias:` option. (being the same `xmpp.` config i am looking for if featured in the release notes)
https://github.com/processone/ejabberd/releases/tag/25.07

**Comment by badlop** [comment]
> it could just pick up/add the domains from hosts_alias: as well, that feels perfect really.

Ok, here is an experimental implementation:

<details>

```diff
diff --git a/src/ejabberd_acme.erl b/src/ejabberd_acme.erl
index 8b16fc727..9adddf365 100644
--- a/src/ejabberd_acme.erl
+++ b/src/ejabberd_acme.erl
@@ -457,9 +457,13 @@ delete_obsolete_data() ->
 get_commands_spec() ->
     [#ejabberd_commands{name = request_certificate, tags = [acme],
 			desc = "Requests certificates for all or some domains",
-			longdesc = "Domains can be `all`, or a list of domains separared with comma characters",
+			longdesc = "Domains is a list of domains separared with comma characters. "
+			           "If domains is `all`, it requests for all domains defined in "
+			           "toplevel option `hosts` and other routes served by ejabberd. "
+			           "If domains is `all_and_alias`, it also requests for domains "
+			           "defined in toplevel option `hosts_alias`.",
 			module = ?MODULE, function = request_certificate,
-			args_desc = ["Domains for which to acquire a certificate"],
+			args_desc = ["Domains for which to acquire a certificate, or `all`, or `all_and_alias`"],
 			args_example = ["example.com,domain.tld,conference.domain.tld"],
 			args = [{domains, string}],
 			result = {res, restuple}},
@@ -490,6 +494,13 @@ request_certificate(Arg) ->
 		      Domains ->
 			  gen_server:call(?MODULE, {request, Domains}, ?CALL_TIMEOUT)
 		  end;
+	      [<<"all_and_alias">>] ->
+		  case auto_domains() of
+		      [] -> {error, no_auto_hosts};
+		      Domains ->
+			  Alias = alias_domains(),
+			  gen_server:call(?MODULE, {request, Domains ++ Alias}, ?CALL_TIMEOUT)
+		  end;
 	      [_|_] = Domains ->
 		  case lists:dropwhile(
 			 fun(D) ->
@@ -585,6 +596,15 @@ auto_domains() ->
 	      not is_ip_or_localhost(Host)
       end, all_domains()).
 
+alias_domains() ->
+    lists:filtermap(
+      fun({HostAlias, _}) ->
+            case check_idna([HostAlias]) of
+                {ok, _} -> {true, HostAlias};
+                {error, _} -> false
+            end
+      end, ejabberd_option:hosts_alias()).
+
 -spec directory_url() -> binary().
 directory_url() ->
     maps:get(ca_url, ejabberd_option:acme(), default_directory_url()).
```

</details>

Now configure:
```yaml
hosts:
  - domain.com

hosts_alias:
  xmpp.domain.com: domain.com
  jabber.domain.com: domain.com
  "exa*": domain.com
```

and request certificates:
```bash
ejabberdctl request_certificate all_and_alias
```

This should request for: domain.com, xmpp.domain.com and jabber.domain.com
